ios/Classes/FilePickerPlugin.m

```diff
@@ -349,27 +349,33 @@ - (void)documentPicker:(UIDocumentPickerViewController *)controller
     if(_result == nil) {
         return;
     }
-    NSMutableArray<NSURL *> *newUrls = [NSMutableArray new];
-    for (NSURL *url in urls) {
-        // Create file URL to temporary folder
-        NSURL *tempURL = [NSURL fileURLWithPath:NSTemporaryDirectory()];
-        // Append filename (name+extension) to URL
-        tempURL = [tempURL URLByAppendingPathComponent:url.lastPathComponent];
-        NSError *error;
-        // If file with same name exists remove it (replace file with new one)
-        if ([[NSFileManager defaultManager] fileExistsAtPath:tempURL.path]) {
-            [[NSFileManager defaultManager] removeItemAtPath:tempURL.path error:&error];
+    NSMutableArray<NSURL *> *newUrls;
+    if(controller.documentPickerMode == UIDocumentPickerModeOpen) {
```

Comment:
The `documentPickerMode` is deprecated. Could you rewrite this to only use the deprecated API up to iOS 14 ?

See https://developer.apple.com/documentation/uikit/uidocumentpickerviewcontroller#1658475

Reply:
Sorry, I don't have a mac device now, so I can't rewrite and test it. 
In addition, I noticed that the deprecated API was also used in the [previous code](https://github.com/miguelpruivo/flutter_file_picker/blob/dff1dde7277b03ac286a3957503731c23a2e4d53/ios/Classes/FilePickerPlugin.m#L169). Maybe this also needs to be migrated to the new API, but I am not familiar with objc.
BTW, on my own ios device, when using the `init(forOpeningContentTypes:)` swift api, I can directly get the absolute path of file/directory. Why doesn't file_picker use this api, but makes a cache file?

Reply:
I do not now why file_picker uses a cache for that, maybe @miguelpruivo knows?

Reply:
> I do not now why file_picker uses a cache for that, maybe @miguelpruivo knows?

There are 2 approaches when opening descriptors for files within app's container
1. Opening a resource directly to the file through URI/UTI or;
2. Caching a copy of the file so it can be manipulated individually within the app and without any constraint regarding the original file;

2 is typically the preferred option since most of the apps use the files for either manipulating (editing) or, even if it's just for sending somewhere, it doesn't have any consequence. If I recall, this was the original approach from the Flutter's image_picker so I followed the same path.

Reply:
@navaronbracke @miguelpruivo 
Please pay attention to #1412 . This bug is very serious on iOS and will endanger user data security.
It has been half a year since I raised the issue, but this dangerous bug still exists and has not been fixed.
This PR is made based on the current implementation, I know that `UIDocumentPickerMode` is a deprecated api, but the code in this library always uses this deprecated api when creating `UIDocumentPickerViewController`.
In this case, why is this PR banned because I use the same deprecated API? (In fact, another PR  #1452 that was approved also used the deprecated `UIDocumentPickerMode` api).
I think a more reasonable solution is to first pass this PR to solve the BUG instead of banning this PR because of an deprecated api that already exists in the library.
Please allow me to repeat it again: this bug is very serious, it will delete files without the user knowing! Please don't keep this bug alive for longer.